MekHQ/src/mekhq/gui/dialog/PartsReportDialog.java

```diff
@@ -127,6 +130,43 @@ public void actionPerformed(ActionEvent e) {
                 refreshOverviewSpecificPart(row, piu, partToBuy);
             }
         };
+
+        Action sell = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                int row = Integer.parseInt(e.getActionCommand());
+                PartInUse piu = overviewPartsModel.getPartInUse(row);
+                campaign.getWarehouse().getSpareParts().stream().filter(p ->
+                    Objects.equals(p.getName(), piu.getName()))
+                    .findFirst()
+                    .ifPresent(p -> campaign.getQuartermaster().sellPart(p, 1));
+                refreshOverviewPartsInUse();
+            }
+        };
+
+        Action sellInBulk = new AbstractAction() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                int row = Integer.parseInt(e.getActionCommand());
+                PartInUse piu = overviewPartsModel.getPartInUse(row);
+                campaign.getWarehouse().getSpareParts().stream().filter(p ->
+                                Objects.equals(p.getName(), piu.getName()))
```

Comment:
I see you're comparing Strings here, is it not possible to compare `part` instead? That way we wouldn't need to pass it into a String.

Reply:
So I thought I tried using IDs earlier and it didn't work, but it turns out I was wrong (or I understand the parts API much better than I did this afternoon). I've refactored this to use the existing `checkForExistingSparePart()` method and IDs instead of string names and it's much cleaner. Thanks for the suggestion!